src/main/java/org/xbill/DNS/ResolverClientFactory.java

```diff
@@ -0,0 +1,13 @@
+package org.xbill.DNS;
+
+/**
+ * Interface for creating the TCP/UDP factories
+ * necessary for the simple resolver.
+ */
+public interface ResolverClientFactory {
+
+  TcpResolverClient createTcpClient();
```

Comment:
Given that this might be an expensive operation, it's not clear that this method is called for every query. Maybe `getTcpClient()` would be more appropriate? OTOH `create` is more natural for a factory, but then the SimpleResolver should only call this once (e.g. in the constructor).

Reply:
I am not sure who should have the responsibility here and if it should be the same for UDP. My thought was that the factory would handle the pools and that create might be a bit of a misnomer. Is `createOrGet` overthinking it?

Reply:
createOrGet is fine.

Reply:
That's fine.